Frontend/src/app/components/product-details/product-details.component.spec.ts

```diff
@@ -1,18 +1,33 @@
 import {ComponentFixture, TestBed} from '@angular/core/testing';
 
 import {ProductDetailsComponent} from './product-details.component';
-import {RouterTestingModule} from "@angular/router/testing";
-import {AbstractMockObservableService} from "../../mocks/mock.service";
-import {ApiService} from "../../services/api.service";
-import {ChartComponent, NgApexchartsModule} from "ng-apexcharts";
+import {RouterTestingModule} from '@angular/router/testing';
+import {AbstractMockObservableService} from '../../mocks/mock.service';
+import {ApiService} from '../../services/api.service';
+import {ChartComponent, NgApexchartsModule} from 'ng-apexcharts';
+import {By} from '@angular/platform-browser';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mueller-Patrick/Betterzon/pullRequest?prid=7401082)
```suggestion
import {By} from "@angular/platform-browser";
```